tests/obs.py

```diff
@@ -479,41 +489,63 @@ def request_search(self, request, uri, headers):
             response = (200, headers, result)
         except Exception as e:
             if DEBUG:
-                print uri, e
+                print(uri, e)
 
         if DEBUG:
-            print 'SEARCH REQUEST', uri, response
+            print('SEARCH REQUEST', uri, response)
 
         return response
 
     #
     # /source/
     #
 
-    @GET(re.compile(r'/source/openSUSE:Factory:Staging/_attribute/openSUSE:LockedBy'))
-    def source_staging_lock(self, request, uri, headers):
-        """Return staging lock."""
-        response = (404, headers, '<result>Not found</result>')
-        try:
-            lock = self.lock if self.lock else self._fixture(uri)
-            response = (200, headers, lock)
-        except Exception as e:
-            if DEBUG:
-                print uri, e
+    @GET(re.compile(r'/source/[\w:]+/_attribute/[\w:]+$'))
+    def source_attribute(self, request, uri, headers):
+        """Return attribute."""
+
+        match = re.search(r'/source/([\w:]+)/_attribute/(\w+):(\w+)$', uri)
+        project = match.group(1)
+        ns = match.group(2)
+        name = match.group(3)
+
+        # never returns 404 - unless the project does not exist,
+        # which we can't (and don't need) to model
+        response = (200, headers, '<attributes/>')
+        print(self.attributes, project, ns, name)
+        if not project in self.attributes:
+            return response
+        hash = self.attributes[project]
+
+        if not ns in hash and not name in hash[ns]:
+            return response
+
+        root = ET.fromstring('<attributes><attribute name="" namespace="">' +
+                             '<value/></attribute></attributes>')
+        root.find('./attribute').set('name', name)
+        root.find('./attribute').set('namespace', ns)
+        root.find('./attribute/value').text = hash[ns][name]
+
+        response = (200, headers, ET.tostring(root))
```

Comment:
May be of interest in future, but could write against dockerized OBS that is already available with a test hardness and tests using it. Avoids needed to rewrite OBS.

Reply:
Actually the docker-compose of OBS works quite nicely - we'd need to fill the database with Factory data though. But this is required anyway soonish to test https://github.com/openSUSE/open-build-service/pull/5140 properly

Reply:
`obs_clone.py` is already doing that. It does not copy everything (I have local unifinish attribute copy), but it already provided skeleton project with a devel package.